auth0/resource_auth0_role.go

```diff
@@ -0,0 +1,167 @@
+package auth0
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"gopkg.in/auth0.v1"
+	"gopkg.in/auth0.v1/management"
+)
+
+func newRole() *schema.Resource {
+	return &schema.Resource{
+
+		Create: createRole,
+		Update: updateRole,
+		Read:   readRole,
+		Delete: deleteRole,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"role_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_ids": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				ForceNew: true,
+			},
+			"permissions": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"resource_server_identifier": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Optional: true,
+			},
+		},
+	}
+}
+
+func createRole(d *schema.ResourceData, m interface{}) error {
+	c := buildRole(d)
+	api := m.(*management.Management)
+	if err := api.Role.Create(c); err != nil {
+		return err
+	}
+
+	users := buildUsers(d)
```

Comment:
It would be good to call `buildUsers` from `buildRole` and abstract the "building".

Reply:
Hm here I'm not really convinced that to do, I tried to change this step but at the end, I would keep it or renamed this steps to call a `assignUsers` and `assignPermissions` to simplify the `createRole` and hide the two API calls.

The problem I faced was if I change this within the `buildRole` that within the update the method `assignUsers` & `assignPermission` is called again and here I definitely won't this step. Because otherwise, I have to check are there old or new `users` & `permissions` available, so the easiest solution is here `ForceNew` on `permissions` and `user_ids`.